web/src/lib/stores/assets-store.svelte.ts

```diff
@@ -1335,12 +1446,111 @@ export class AssetStore {
     }
   }
 
-  async getNextAsset(asset: AssetResponseDto): Promise<AssetResponseDto | undefined> {
-    let bucket = await this.#getBucketInfoForAsset(asset);
+  async getClosestAssetToDate(date: DateTime) {
+    let bucket = this.#findBucketForDate(date);
     if (!bucket) {
       return;
     }
+    await this.loadBucket(bucket.bucketDate, { cancelable: false });
+    const asset = bucket.findClosest(date);
+    if (asset) {
+      return asset;
+    }
+
+    let bucketIndex = this.buckets.indexOf(bucket) + 1;
+    while (bucketIndex < this.buckets.length) {
+      bucket = this.buckets[bucketIndex];
+      await this.loadBucket(bucket.bucketDate);
+      const next = bucket.dateGroups[0]?.intersetingAssets[0]?.asset;
+      if (next) {
+        return next;
+      }
+      bucketIndex++;
+    }
+  }
+
+  async getNextAsset(
+    idable: { id: string },
+    skipTo: 'asset' | 'day' | 'month' | 'year' = 'asset',
+  ): Promise<AssetResponseDto | undefined> {
+    const bucket = this.#findBucketForAsset(idable.id);
+    if (!bucket) {
+      return;
+    }
+    const asset = bucket.findById(idable.id);
+    if (!asset) {
+      return;
+    }
+
+    switch (skipTo) {
+      case 'day': {
+        return this.#getNextDay(asset, bucket);
+      }
+      case 'month': {
+        return this.#getNextMonth(asset, bucket);
+      }
+      case 'year': {
+        return this.#getNextYear(asset, bucket);
+      }
+      case 'asset': {
+        return this.#getNextAsset(asset, bucket);
+      }
+    }
+  }
+
+  async #getNextDay(asset: AssetResponseDto, bucket: AssetBucket) {
+    let prevDay = DateTime.fromISO(asset.localDateTime).toUTC().get('day') - 1;
+    const bucketIndex = this.buckets.indexOf(bucket);
+
+    let prevDayGroup;
+    while (prevDay >= 0) {
+      prevDayGroup = bucket.findDateGroupByDay(prevDay);
+      if (prevDayGroup) {
+        break;
+      }
+      prevDay--;
+    }
+    if (prevDayGroup === undefined) {
+      let nextBucketIndex = bucketIndex + 1;
+      while (nextBucketIndex < this.buckets.length) {
+        const otherBucket = this.buckets[nextBucketIndex];
+        await this.loadBucket(otherBucket.bucketDate, { cancelable: false });
+        const next = otherBucket.dateGroups[0]?.intersetingAssets[0]?.asset;
+        if (next) {
+          return next;
+        }
+        nextBucketIndex++;
+      }
+    } else {
+      return prevDayGroup.intersetingAssets.at(0)?.asset;
+    }
```

Comment:
Flip these conditions to avoid else case and reduce the amount of code that's indented.